### PR TITLE
chore(operations): Reduce number of CI jobs

### PR DESCRIPTION
**Pull request by Hoverbear**
Compress a lot of our CI jobs into fewer jobs.

### CODE REVIEW

File: .github/workflows/lints.yml [hunk 1]
```diff
@@ -104,86 +64,3 @@ jobs:
...
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - run: bash scripts/environment/bootstrap-macos-10.sh
-      - run: bash scripts/environment/prepare.sh
-      - run: make slim-builds
-      - run: make check
-
-  check-windows:
```
Comment (jszwedko) [inline .github/workflows/lints.yml:172]: I take it we don't need to do these checks on Windows and OSX?

File: .github/workflows/lints.yml [hunk 2]
```diff
@@ -71,22 +39,14 @@ jobs:
...
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - run: sudo bash scripts/environment/bootstrap-ubuntu-20.04.sh
-      - run: bash scripts/environment/prepare.sh
-      - run: make slim-builds
```
Comment (jszwedko) [inline .github/workflows/lints.yml:88]: Would it make sense to still use `slim-builds` for this check?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -288,9 +288,9 @@ jobs:
           - "v1.11.0" # https://github.com/kubernetes/minikube/issues/8799
         kubernetes_version:
           - "v1.18.6"
-          - "v1.17.9"
-          - "v1.16.12" # v1.16.13 is broken, see https://github.com/kubernetes/minikube/issues/8840
-          - "v1.15.12"
+          #- "v1.17.9"
```
Comment (jszwedko) [inline .github/workflows/tests.yml:291]: I assume this'll be augmented by https://github.com/timberio/vector/issues/3338 ?

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -116,7 +116,7 @@ jobs:
       - run: bash scripts/environment/prepare.sh
       - run: make slim-builds
       - run: make test-shutdown
-      # Currently has too few tests to justify a separate task in CI.
+      - run: make test-behavior
```
Comment (jszwedko) [inline .github/workflows/tests.yml:119]: It seems like this is also run by `test-behavior`. Do we need it here too?

File: .github/workflows/lints.yml (hunk 1 above)
Comment (Hoverbear) [reply .github/workflows/lints.yml:172]: `check` is a subset of `test` which is already run, so yes.

File: .github/workflows/tests.yml (hunk 4 above)
Comment (Hoverbear) [reply .github/workflows/tests.yml:119]: Whoops I grabbed the wrong one! Nice catch! omg!

File: .github/workflows/tests.yml (hunk 3 above)
Comment (Hoverbear) [reply .github/workflows/tests.yml:291]: Yeah! I just disabled the mid-versions for now. I'm not sure that's a safe bet, but it'll ease off a lot of load. I think #3338 is doing to augment/resolve this.

File: distribution/nix/default.nix.erb [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Taken from https://github.com/nixos/nixpkgs
+I # Taken from https://github.com/nixos/nixpkgs
```
Comment (jszwedko) [inline distribution/nix/default.nix.erb:1]: :eyes: Should this be here?

Comment (Hoverbear) [reply distribution/nix/default.nix.erb:1]: Definitely not...

File: .github/workflows/lints.yml (hunk 2 above)
Comment (jszwedko) [reply .github/workflows/lints.yml:88]: Doh, missed that :smile:

**Pull request merged**